### PR TITLE
Auto resize example iframes when the content changes

### PR DESCRIPTION
**Pull request by alex-ju**
This PR uses [iframe-resizer](https://github.com/davidjbradshaw/iframe-resizer) (a solution based on MutationObserver with fallbacks and polyfill for old browsers) to adjust the height of the iframe based on changes to the height of the content

Note to reviewer:
For example, when the `details` component is interacted with the revealed content is obscured as the iframe does not adjust to accommodate it. You can find a details component in the [Fill in a form](http://localhost:3000/patterns/fill-in-a-form/) pattern page.

[Trello card](https://trello.com/c/T2559vF1)

### CODE REVIEW

**Comment by 36degrees** [comment]
Is there a reason for vendor-ing the javascript file rather than including it as an dependency in npm?

**Comment by alex-ju** [comment]
Mmm.. not really. We can use NPM for this one, indeed.

**Comment by 36degrees** [comment]
That being the case, do we want to merge this as-is and add a card to move to using packages rather than vendoring?

**Comment by alex-ju** [comment]
Yep, I can write the card for that

**Comment by alex-ju** [comment]
Updated with the commit to use `iframe-resizer` via npm, will make a card and PR to consume the rest of the scripts via `npm`.

File: src/javascripts/vendor/iframeResizer.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/*! iFrame Resizer (iframeSizer.min.js ) - v3.5.15 - 2017-10-15
```
Comment (36degrees) [inline src/javascripts/vendor/iframeResizer.js]: Is this file meant to still be committed?

File: views/layouts/layout-example.njk [hunk 2]
```diff
@@ -6,18 +6,21 @@
...
     {% for stylesheet in stylesheets %}
     <link href="{{ stylesheet }}" rel="stylesheet" media="all" />
     {%- endfor %}
 
-    <script src="/javascripts/application.js"></script>
+    <!--[if lt IE 9]>
+      <script src="/javascripts/ie.js"></script>
+    <![endif]-->
+    <script src="/javascripts/vendor/modernizr.js"></script>
+    {% include "_tracking-head.njk" %}
```
Comment (36degrees) [inline views/layouts/layout-example.njk]: Is this meant to be included as part of this change?

File: src/javascripts/vendor/iframeResizer.js (hunk 1 above)
Comment (alex-ju) [reply src/javascripts/vendor/iframeResizer.js]: Nope, I shall remove it.

File: views/layouts/layout-example.njk (hunk 2 above)
Comment (alex-ju) [reply views/layouts/layout-example.njk]: I can make a separate PR for that. Looks like we didn't add tracking on the example pages, which I thought might be useful.

Comment (36degrees) [reply views/layouts/layout-example.njk]: Suggest doing as a separate PR

**Comment by nickcolley** [comment]
What browsers / devices has this been tested against?

**Comment by alex-ju** [comment]
Tested with:
IE9, 10, 11 on Win
Firefox, Chrome, Safari latest versions on OSX
Desktop only. No small devices.

File: src/javascripts/components/example.js [hunk 3]
```diff
@@ -6,9 +6,7 @@
   GOVUK.example = {
     init: function (selector) {
       // Example iframe; set the height equal to the body height
-      $(selector).on('load', function () {
-        this.style.height = this.contentDocument.body.scrollHeight + 'px'
-      })
+      $(selector).iFrameResize({scrolling: 'auto'})
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply src/javascripts/components/example.js]: Updated by wrapping it with a `try-catch` block.

File: src/javascripts/components/example.js [hunk 4]
```diff
@@ -5,10 +5,14 @@
...
-      // Example iframe; set the height equal to the body height
-      $(selector).on('load', function () {
-        this.style.height = this.contentDocument.body.scrollHeight + 'px'
-      })
+      try {
+        // Example iframe; set the height equal to the body height
+        $(selector).iFrameResize({scrolling: 'auto'})
+      } catch (err) {
+        if (err) {
+          console.log(err.message)
```
Comment (joelanman) [inline src/javascripts/components/example.js]: maybe use `console.error` for this?

Comment (alex-ju) [reply src/javascripts/components/example.js]: Updated

**Comment by alex-ju** [comment]
@hannalaakso the details component doesn't have the polyfill; we don't have yet a system in place (or none that I'm aware of) to copy other assets than sass from the `node_modules` to the project assets.

**Comment by alex-ju** [comment]
Thanks, @hannalaakso!

**Pull request merged**